src/Pact/Eval.hs

```diff
@@ -643,11 +643,15 @@ computeUserAppGas :: Def Ref -> Info -> Eval e Gas
 computeUserAppGas Def{..} ai = computeGas (Left (ai, asString _dDefName)) GUserApp
 
 -- | prepare reduced args and funtype, and typecheck
-prepareUserAppArgs :: Def Ref -> [Term Ref] -> Eval e ([Term Name], FunType (Term Name))
-prepareUserAppArgs Def{..} as = do
-  as' <- mapM reduce as
+prepareUserAppArgs :: Def Ref -> [Term Ref] -> Info -> Eval e ([Term Name], FunType (Term Name))
```

Comment:
this info is an unnecessary change. `Def` already has an `Info` field, and you can just pass the def object to `evalError'` or pass the info directly. We've already opened the record up.

Reply:
But isn't that the def for the function's definition, which would not contain the `Info` at the application site? Where the error is occurring at the application site.

Reply:
Ah yeah, good point. I missed the subtlety.

Reply:
np -- ty for the review!